samples/client/petstore/java/src/main/java/io/swagger/client/ApiClient.java

```diff
@@ -237,6 +238,67 @@ public String parameterToString(Object param) {
     }
   }
 
+  /*
+    Format to {@code Pair} objects.
+  */
+  public List<Pair> parameterToPairs(String collectionFormat, String name, Object value){
+    List<Pair> params = new ArrayList<Pair>();
+
+    // preconditions
+    if (name == null || name.isEmpty()) return params;
+
+    if (value == null) {
+      params.add(new Pair(name, ""));
+      return params;
+    }
```

Comment:
the previous logic was to ignore the parameter if its value is `null`, which means the parameter is optional (as required parameter's value should have been checked and ensured to not be `null`.)